lib/prawn/attachment.rb

```diff
@@ -54,16 +53,34 @@ def attach(src, options = {})
 
     private
 
-    def data_and_opts(path, src, options)
-      opts = options.dup
-      return [src, opts] unless path.file?
+    def file?(path)
+      return false if path.include?("\u0000")
```

Comment:
It's not clear at all why this is important or in what cases this might happen. There are no tests which could be used to replicate the issue either. Either way, surely the `File.file?` call below would check to ensure the file exists or raise an error.

Reply:
There are files (like `.pdf`) that have null characters and be non-empty. If the null char is present, it's not a file path. I was thinking about a specific `path:` key argument to distinguish a string from a file path.